test/steward_multi_patronage.js

```diff
@@ -0,0 +1,139 @@
+    const collectPatronageT10_tx = await steward._collectPatronage(testTokenId1);
+    const benefactorFundsT10 = await steward.benefactorFunds.call(accounts[8])
+    const collectPatronageT10BlockTime = (await web3.eth.getBlock(collectPatronageT10_tx.receipt.blockNumber)).timestamp
+    const lastCollectedUserT10 = await steward.timeLastCollectedUser.call(accounts[2])
+
+    // Check patronage after 10mins is correct
+    const patronDepositAfter10min = await steward.deposit.call(accounts[2]);
+    const expectedPatronageAfter10min = multiPatronageCalculator('600',
+      [{ patronageNumerator: testToken1.patronageNumerator.toString(), price: priceOfToken1.toString() }])
+    assert.equal(patronDepositInitial.toString(), patronDepositAfter10min.add(expectedPatronageAfter10min).toString());
+    assert.equal(collectPatronageT10BlockTime.toString(), lastCollectedUserT10.toString())
+    assert.equal(expectedPatronageAfter10min.toString(), benefactorFundsT10.toString())
+
+    /////////////////// TIME = 20 ////////////////////
+    //////////////////////////////////////////////////
+    //////////////////////////////////////////////////
+    await time.increase(time.duration.minutes(10));
+    // await waitTillBeginningOfSecond()
+
+    // Buy a 2nd token
+    const buyToken2Tx = await steward.buy(testTokenId2, web3.utils.toWei('2', 'ether'), { from: accounts[2], value: web3.utils.toWei('1', 'ether') });
+    const buyToken2BlockTime = (await web3.eth.getBlock(buyToken2Tx.receipt.blockNumber)).timestamp
+    const lastCollectedUserT20 = await steward.timeLastCollectedUser.call(accounts[2])
+    const priceOfToken2 = await steward.price.call(testTokenId2)
+    assert.equal(buyToken2BlockTime.toString(), lastCollectedUserT20.toString())
+
+    const patronDepositAfter20min = await steward.deposit.call(accounts[2]);
+    const patronDepositCalculatedAfter20min = await steward.depositAbleToWithdraw.call(accounts[2]);
+    const expectedPatronage10MinToken1 = multiPatronageCalculator('600',
+      [{ patronageNumerator: testToken1.patronageNumerator.toString(), price: priceOfToken1.toString() }])
+
+    assert.equal(patronDepositAfter20min.toString(), patronDepositAfter10min.sub(expectedPatronage10MinToken1).add(ether('1')).toString());
+    assert.equal(patronDepositCalculatedAfter20min.toString(), patronDepositAfter10min.add(ether('1')).sub(expectedPatronage10MinToken1).toString());
+
+    /////////////////// TIME = 30 ////////////////////
+    //////////////////////////////////////////////////
+    //////////////////////////////////////////////////
+    await time.increase(time.duration.minutes(10));
+    await waitTillBeginningOfSecond()
+
+    await steward._collectPatronage(testTokenId1);
+
+    const patronDepositAfter30min = await steward.deposit.call(accounts[2]);
+    const patronDepositCalculatedAfter30min = await steward.depositAbleToWithdraw.call(accounts[2]);
+    const expectedPatronageMulti = multiPatronageCalculator('601',
```

Comment:
Good to note that it is 601 seconds because the waitTillBeginningOfSecond function is called, therefore adding another second.

Reply:
Agreed